packages/gatsby-plugin-page-creator/src/gatsby-node.js

```diff
@@ -48,48 +49,56 @@ exports.createPagesStatefully = async (
     )
   }
 
-  const pagesDirectory = systemPath.posix.join(pagesPath)
-  const pagesGlob = `${pagesDirectory}/**/*.{${exts}}`
+  const pagesDirectory = systemPath.resolve(process.cwd(), pagesPath)
+  const pagesGlob = `**/*.{${exts}}`
 
   // Get initial list of files.
-  let files = await glob(pagesGlob)
-  files.forEach(file => _createPage(file, pagesDirectory, createPage))
+  let files = await glob(pagesGlob, { cwd: pagesPath })
```

Comment:
could you move to `micromatch` here as well? or move to glob everwhere. It doesn't make much sense dependency wise to have 2 glob deps 😄

Reply:
`micromatch` is a replacement for `minimatch`; it doesn't include the filesystem parts of `glob`. [`glob-fs`](https://www.npmjs.com/package/glob-fs) exists based on `micromatch` but according to [the github page](https://github.com/micromatch/glob-fs) it's experimental and doesn't work on Windows.

Perhaps declaring a dependency on `minimatch` instead of `micromatch` in the other places would be a good way to go.

Reply:
Hmm... chokidar uses micromatch (via anymatch), but as paulmelnikow says, micromatch doesn't provide any filesystem utility so we'd need to initialize the files array some other way.

**gatsby-source-filesystem** handles the initialization by using chokidar with a state machine (xstate). Would you prefer that over a glob dependency? It'd be a bit of a refactor.

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-filesystem/src/gatsby-node.js

Reply:
TIL @paulmelnikow you can ignore this 👍

@Warped2713 we could go for chokidar to listen to pages but probably not worth it unless this one is getting slow. Chokidar is much more memory hungry. We might want to change this in the future.